packages/std/src/ibc.rs

```diff
@@ -648,36 +661,56 @@ pub struct IbcReceiveResponse<T = Empty> {
     pub events: Vec<Event>,
 }
 
-// Custom implementation in order to implement it for all `T`, even if `T` is not `Default`.
-impl<T> Default for IbcReceiveResponse<T> {
-    fn default() -> Self {
-        IbcReceiveResponse {
-            acknowledgement: Binary(vec![]),
+impl<T> IbcReceiveResponse<T> {
+    /// Create a new response with the given acknowledgement.
+    ///
+    /// ## Examples
+    ///
+    /// ```
+    /// use cosmwasm_std::{StdAck, Binary, IbcReceiveResponse};
+    ///
+    /// let _: IbcReceiveResponse = IbcReceiveResponse::new(StdAck::success(b"my ack"));
+    /// let _: IbcReceiveResponse = IbcReceiveResponse::new(Binary::from(b"content"));
+    /// ```
+    pub fn new(ack: impl Into<Binary>) -> Self {
+        Self {
+            acknowledgement: Some(ack.into()),
             messages: vec![],
             attributes: vec![],
             events: vec![],
         }
     }
-}
 
-impl<T> IbcReceiveResponse<T> {
-    pub fn new() -> Self {
-        Self::default()
+    /// Creates a new response without an acknowledgement.
+    ///
+    /// This allows you to send the acknowledgement asynchronously later using [`IbcMsg::WriteAcknowledgement`].
+    /// If you want to send the acknowledgement immediately, use [`IbcReceiveResponse::new`].
+    pub fn without_ack() -> Self {
+        Self {
+            acknowledgement: None,
+            messages: vec![],
+            attributes: vec![],
+            events: vec![],
+        }
     }
 
     /// Set the acknowledgement for this response.
     ///
+    /// If you provide `None`, the acknowledgement is asynchronous and you can provide it later
+    /// using [`IbcMsg::WriteAcknowledgement`].
+    ///
     /// ## Examples
     ///
     /// ```
     /// use cosmwasm_std::{StdAck, IbcReceiveResponse};
     ///
     /// fn make_response_with_ack() -> IbcReceiveResponse {
     ///     let ack = StdAck::success(b"\x01"); // 0x01 is a FungibleTokenPacketSuccess from ICS-20.
-    ///     IbcReceiveResponse::new().set_ack(ack)
+    ///     IbcReceiveResponse::new(ack)
     /// }
+    /// _ = make_response_with_ack();
     /// ```
-    pub fn set_ack(mut self, ack: impl Into<Binary>) -> Self {
+    pub fn set_ack(mut self, ack: impl Into<Option<Binary>>) -> Self {
```

Comment:
Let's remove this function. There is no need to change the acknowledgement after contruction.